### DISCUSSION TITLE
GitHub should become COSS (open core model)

### DISCUSSION THREAD

**Discussion by suprstarrd** [community/community#15199] [Other Feature Feedback, Questions, & Ideas] [answered]
**EDIT**: I am now converting this discussion (originally titled "Why is GitHub not open source?") into a request for GitHub to become [COSS](https://en.wikipedia.org/wiki/Open-core_model). GitHub has understandable (although potentially questionable) reasons to not be fully open source, but GitHub's core being open source lets users have more trust in the service. 

GitHub's competitors have done an amazing job with this model; GitHub can do even better if they were to follow suit. It may not get GitHub.com more users, but it will get *GitHub* more users *and more sales* if done properly. GitHub becoming COSS feels like a natural step.

<hr>

**ORIGINAL POST**: Why GitHub isn't at least *partially* open source? I can understand it not being fully open source / source available, but GitHub and others have proven that it is possible to make money while being a fully open source product. While GitHub does have some of its offspring as open source code, I'm surprised that GitHub's core isn't. It feels odd that GitHub isn't being the very thing it represents to such a massive degree.

**Comment by micwoj92** [comment]
It's because proprietary has PRO in it.

**Comment by suprstarrd** [comment]
Oh look, this got a label! ❤️

This still doesn't have an answer, unfortunately. I imagine that at this point the hole's been dug too deep to get out of (although, if it's not too deep then treat this as a feature request: please make GitHub at least partially open source), but I'm sure there was some reason why this wasn't open source from the start.

**Comment by Arastoi** [comment]
Tancs

در تاریخ ۲۱ آوریل ۲۰۲۲ ۵:۳۳ ب.ظ، "JBMagination" ***@***.***>
نوشت:

**Reply by suprstarrd** [reply]
What?

**Comment by suprstarrd** [comment, marked as answer]
**This post answers the original question.**

I decided to do some digging, and found one of the original founders of GitHub, [@](https://github.com/mojombo)[mojombo](https://github.com/mojombo)! [This blog post](https://tom.preston-werner.com/2008/10/18/how-i-turned-down-300k.html) discusses how GitHub came to be in detail.

But more importantly, [this blog post](https://tom.preston-werner.com/2011/11/22/open-source-everything.html) from the same person discusses why GitHub's core isn't open source *(emphasis in italics is my own)*: 
> **Ok, then what shouldn’t I open source?**
> That’s easy. Don’t open source anything that represents core business value.
> 
> Here are some examples of what we don’t open source and why:
> * Core GitHub Rails app  *(easier to sell when closed)*
> * The Jobs Sinatra app (specially crafted integration with github.com)
> 
> Notice that everything we keep closed has specific business value that could be compromised by giving it away to our competitors. Everything we open is a general purpose tool that can be used by all kinds of people and companies to build all kinds of things.

GitHub's core isn't open source because it's business. 

With that said, I do have a rebuttal as to why it should still be open source. *Excluding the fact that GitLab exists,* the difference nowadays is that **GitHub has now proven it's possible to make money while still being open source thanks to GitHub Sponsors and the like.** On top of that, part of GitHub Enterprise is exclusive to Enterprise, and isn't a part of GitHub.com.

On the other hand, open sourcing can actually be business by itself. I know of businesses that thrive solely because they're open source (and in some cases copyleft); if they weren't, they would fail. GitHub is NOT one of those businesses, but I do think open sourcing GitHub's core could be a great push for not just businesses but people in general who are already on GitHub to consider open sourcing more of their code.

**Reply by Snailedlt** [reply]
GitHub has the large majority of open source projects and enthusiasts on their platform, which means they would also get a ton of useful Pull Requests for features they otherwise wouldn't have time to implement themselves.

I think in terms of overall software quality, open sourcing would definitely help. Especially things like the GitHub UI, which tons of people interact with every day

**Comment by suprstarrd** [comment]
I am now converting this discussion (originally titled "Why is GitHub not open source?") into a request for GitHub to become [COSS](https://en.wikipedia.org/wiki/Open-core_model). GitHub has understandable (although potentially questionable) reasons to not be fully open source, but GitHub's core being open source lets users have more trust in the service. 

GitHub's competitors have done an amazing job with this model; GitHub can do even better if they were to follow suit. It may not get GitHub.com more users, but it will get *GitHub* more users *and more sales* if done properly. GitHub becoming COSS feels like a natural step.
